### PR TITLE
MCR-902 use message properties in orcid javascript

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MIR-902).

### CODE REVIEW

File: mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js [hunk 1]
```diff
@@ -1,73 +1,98 @@
-var orcidStatusURL  = webApplicationBaseURL + "rsc/orcid/status/";
-var orcidPublishURL = webApplicationBaseURL + "rsc/orcid/publish/";
+(function(){
```
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js:1]: Codacy Issue found: [Move the invocation into the parens that contain the function.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111&bid=18083276)

File: mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js [hunk 2]
```diff
@@ -1,73 +1,98 @@
...
+	const orcidStatusURL  = webApplicationBaseURL + "rsc/orcid/status/";
+	const orcidPublishURL = webApplicationBaseURL + "rsc/orcid/publish/";
+	const translateURL = webApplicationBaseURL + "rsc/locale/translate/";
 
-var orcidIcon = "<img alt='ORCID iD' src='" + webApplicationBaseURL + "images/orcid_icon.svg' class='orcid-icon' />";
+	const orcidIcon = "<img alt='ORCID iD' src='" + webApplicationBaseURL + "images/orcid_icon.svg' class='orcid-icon' />";
```
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js:6]: Codacy Issue found: ['webApplicationBaseURL' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111)

File: mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js [hunk 3]
```diff
@@ -1,73 +1,98 @@
...
-});
+	let docReady=false;
+	let langReady=false;
 
-function getORCIDPublicationStatus(div) {
-	var id = jQuery(div).data('id');
-	jQuery.get(orcidStatusURL + id, function(status) {
-		console.log(status);
-		setORCIDPublicationStatus(div, status);
+	jQuery.ajax(translateURL + currentLang +"/" + keyPrefix).done(function(data){
```
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js:19]: Codacy Issue found: ['currentLang' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111)

File: mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js [hunk 4]
```diff
@@ -1,73 +1,98 @@
-var orcidStatusURL  = webApplicationBaseURL + "rsc/orcid/status/";
-var orcidPublishURL = webApplicationBaseURL + "rsc/orcid/publish/";
+(function(){
+	const orcidStatusURL  = webApplicationBaseURL + "rsc/orcid/status/";
+	const orcidPublishURL = webApplicationBaseURL + "rsc/orcid/publish/";
```
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js:3]: Codacy Issue found: ['webApplicationBaseURL' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111)

File: mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js [hunk 5]
```diff
@@ -1,73 +1,98 @@
...
-		setORCIDPublicationStatus(div, status);
+	jQuery.ajax(translateURL + currentLang +"/" + keyPrefix).done(function(data){
+		orcidTextStatusIsInProfileTrue = data[orcidTextStatusIsInProfileTrue];
+		orcidTextStatusIsInProfileFalse = data[orcidTextStatusIsInProfileFalse];
+		orcidTextPublishUpdate = data[orcidTextPublishUpdate];
+		orcidTextPublishCreate = data[orcidTextPublishCreate];
+		orcidTextPublishConfirm = data[orcidTextPublishConfirm];
+	}).always(function () {
+		langReady=true;
+		tryInit();
```
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js:27]: Codacy Issue found: ['tryInit' was used before it was defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111&bid=18083276)

File: mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js [hunk 6]
```diff
@@ -1,73 +1,98 @@
-var orcidStatusURL  = webApplicationBaseURL + "rsc/orcid/status/";
-var orcidPublishURL = webApplicationBaseURL + "rsc/orcid/publish/";
+(function(){
+	const orcidStatusURL  = window["webApplicationBaseURL"] + "rsc/orcid/status/";
+	const orcidPublishURL = window["webApplicationBaseURL"] + "rsc/orcid/publish/";
+	const translateURL = window["webApplicationBaseURL"] + "rsc/locale/translate/";
 
-var orcidIcon = "<img alt='ORCID iD' src='" + webApplicationBaseURL + "images/orcid_icon.svg' class='orcid-icon' />";
+	const orcidIcon = "<img alt='ORCID iD' src='" + webApplicationBaseURL + "images/orcid_icon.svg' class='orcid-icon' />";
```
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js:6]: Codacy Issue found: ['webApplicationBaseURL' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111)

File: mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js (hunk 1 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js:1]: Codacy Issue found: [Move the invocation into the parens that contain the function.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111&bid=18083276)

File: mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js [hunk 7]
```diff
@@ -1,73 +1,99 @@
-var orcidStatusURL  = webApplicationBaseURL + "rsc/orcid/status/";
-var orcidPublishURL = webApplicationBaseURL + "rsc/orcid/publish/";
+(function(){
```
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/mir/mycore2orcid.js:1]: Codacy Issue found: [Move the invocation into the parens that contain the function.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 27cd1986cd5222571d0ba3773db8a9a7e279135f! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2594-pull_request-471

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5519111&bid=18083276)

**Pull request merged**